Rover/mode.cpp

```diff
@@ -123,8 +123,8 @@ void Mode::get_pilot_desired_steering_and_throttle(float &steering_out, float &t
     // check for special case of input and output throttle being in opposite directions
     float throttle_out_limited = g2.motors.get_slew_limited_throttle(throttle_out, rover.G_Dt);
     if ((is_negative(throttle_out) != is_negative(throttle_out_limited)) &&
-        ((g.pilot_steer_type == PILOT_STEER_TYPE_DEFAULT) ||
```

Comment:
This would be a *lot* neater if you made g.pilot_steer_type `AP_Enum<PilotSteerType>`